hyperscript/element.py

```diff
@@ -105,8 +115,17 @@ def __str__(self) -> str:
         opening_tag = " ".join(opening_tags)
         if self.is_void:
             return f"<{opening_tag}>"
-        children = "".join(self._stringify(child) for child in self.children)
-        return f"<{opening_tag}>{children}</{self.tag}>"
+        children = []
+        for child in self.get_children(remove_empty=True):
+            if (
+                self.remove_empty
+                and isinstance(child, Element)
+                and not child.is_void
+                and not child.get_children(remove_empty=True)
```

Comment:
this is an N^2 loop isn't it, is there a reason to do this lazily, or should it be done at init or cached?

Reply:
or could you somehow collect the children, and then stringify each one  in this method without calling get_children twice?

Reply:
Yes, let me see if I can make this more efficient.